classes/class-kp-session.php

```diff
@@ -92,7 +92,7 @@ public function get_session( $order = null ) {
 		}
 
 		// If we have a Klarna session and we get here, we should update it.
-		if ( null !== $this->klarna_session ) {
+		if ( null !== $this->klarna_session && false !== strpos( WC()->session->get( 'chosen_payment_method' ), 'klarna_payments' ) ) {
```

Comment:
WC()->session might be null and will result in an error if you try to access ->get on null. Also `strpos` with the first arg as null is deprecated since 8.2.0 i believe.

I recommend making a helper method that can handle either using the cart or a order to check this same thing, with some more logic built in. Then that can also be reused in the other place you added a similar check.